### PR TITLE
cmd/common: Clean up special event management

### PR DESCRIPTION
**Pull request by blanquicet**
# cmd/common: Clean up special event management

- There is no need to make a copy of the Event to print special events, instead, use pointers.
- Move `GetBaseEvent` into the embedded struct so that it doesn't need to be implemented by every gadget.
- Rename ManageSpecialEvent to HandleSpecialEvent as in this case, "Handle" is more accurate than "Manage".

### CODE REVIEW

File: pkg/types/types.go
Comment (flyth) [inline pkg/types/types.go:97]:
```suggestion
// GetBaseEvent is needed to implement commonutils.BaseElement and
// snapshot.SnapshotEvent interfaces.
func (*e Event) GetBaseEvent() *Event {
	return e
}
```

I think we should use a pointer receiver here as well, otherwise the struct is still being copied.

**Review by flyth — COMMENTED** [review]
Much cleaner this way, thanks!

File: pkg/types/types.go
Comment (blanquicet) [reply pkg/types/types.go:97]: Yeah, that would also help. However, doing so with generics became quite complex. I will let it for another PR.

**Comment by blanquicet** [comment]
Rebased to ensure the CI is all green after merging https://github.com/inspektor-gadget/inspektor-gadget/pull/1141.

**Review by mauriciovasquezbernal — APPROVED** [review]
Nice cleanup. Regarding pointers, I think we should check if this is possible to pass pointers in more places, it seems to me we're making a lot of unneeded copies, but of course this is something to explore on a different PR. 

(I'm approving now but please check that there are some instances of ManageSpecialEvent that need to be updated in cmd/kubectl-gadget/traceloop.go.

**Pull request merged**
